proto/documents/v1alpha/documents.proto

```diff
@@ -165,6 +168,16 @@ message DeletePublicationRequest {
   string document_id = 1;
 }
 
+
+// Request for getting a single publication.
+message PushPublicationRequest {
```

Comment:
Don't we need a gateway address here? Seems without it we end up hardcoding the addresses, and having to do something special depending on whether we are in testnet or not. I believe it would be much less brittle if we just allow the client to specify the gateway address, and get rid of any of these concerns inside mintterd, because it really doesn't care (unless I'm missing something).

I'd suggest adding a field for the gateway address URL. It could be either a web address, domain name, or full multiaddr. We can resolve web address to peer ID using the well-known endpoint if necessary.

Not having this hardcoded could potentially make this code path more testable (we could spin up a gateway locally, and test that push works, for example).

All in all, I believe clients should choose the address they want to push to.
```suggestion
message PushPublicationRequest {
```

Reply:
@burdiyan Resolving the Address via .well-known would critically slow down the whole process, and we want it to be super fast. Not in the case of a successful pushing (I guess we can afford to lose an extra 1-2s) but in the case of failing. Since the frontend is going to be using this endpoint extensively for documents that are already there, we have to be fast on address resolution.

Reply:
I don't see how this is going to be an issue at all. We could cache the well-known result for example. Even without caching, I don't believe an extra 100ms (probably less) round trip would affect anything at all.

Reply:
More importantly, for these two well-known hardcoded gateways, we could ignore the process of resolving the address because we already know them. So only if they would ask for a totally different one we would go and resolve the addresses.